### ISSUE TITLE
Move release process document to version control

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4543]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)| @jml reported|
|-|-|
|Trac ID|trac#4543|
|Type|task|
|Created|2010-07-04 16:01:02Z|
|Branch|https://github.com/twisted/twisted/tree/release-process-4543|

ReleaseProcess should be moved into version control and managed there.

The document on the wiki is now basically complete and probably mostly accurate. Where it's wrong, that's a defect, and where the process can be changed to be better, that's an enhancement.

The length and complexity of the process it describes makes it a near essential to doing the release. It ought to live near all of the other release essentials.

It refers to numerous scripts based in the tree. When these scripts change, the document ought to change as well. 

Attachments:

* [version-control-release-process-4543-1.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/228/228bf8cc11958bfbebaf85e8ff6e11ea5ee3a701/060dafe502efaa6c925303a65a5f241f8ae6e9fb.patch) (20918 bytes) - added by bramwelt on 2014-05-14 06:30:39Z - ReST Version of Release Process

<details><summary>Searchable metadata</summary>

```
trac-id__4543 4543
type__task task
reporter__jml jml
priority__normal normal
milestone__totally_automated_release_infrastructure totally automated release infrastructure
branch__branches_release_process_4543 branches/release-process-4543
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__release_management release management
keywords__None None
time__1278259262000000 1278259262000000
changetime__1457329783769579 1457329783769579
version__None None
owner__hawkowl hawkowl
cc__thijs cc__bramwelt
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_1"></a>|@glyph commented|
|-|-|

If the sphinx conversion is imminent, it might be worthwhile to wait for it.  However, sphinx or lore, how do you propose to convert the document into something suitable for viewing and editing in SVN as opposed to trac markup?  (Are you volunteering to do the translation manually?)

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_2"></a>|@jml commented|
|-|-|

Yes, volunteering to do the transition manually.

Where can I track progress on Sphinx stuff?

Note that switching to Sphinx rather than Lore requires changes to the release process itself, which is one of the reasons I'm keen to have this in the tree.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_3"></a>|@glyph commented|
|-|-|

Replying to [jml](#note_2):
> Yes, volunteering to do the transition manually.
> 
> Where can I track progress on Sphinx stuff?

The best answer I can find is this: http://twistedsphinx.funsize.net/transition_plan.html

Somebody should really bug khorn to turn that into a ticket.

> Note that switching to Sphinx rather than Lore requires changes to the release process itself, which is one of the reasons I'm keen to have this in the tree.

Yes, definitely.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_4"></a>|@jml commented|
|-|-|

I've filed [#4553](https://github.com/twisted/twisted/issues/4553) to be that master ticket.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra commented|
|-|-|

Replying to [jml](#note_4):
> I've filed [#4553](https://github.com/twisted/twisted/issues/4553) to be that master ticket.

See [#4500](https://github.com/twisted/twisted/issues/4500) for the master ticket.

**Comment by twisted-trac** [comment]
|<img alt="bramwelt's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|bramwelt set owner to bramwelt<br>bramwelt set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="bramwelt's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_8"></a>|bramwelt removed owner<br>bramwelt set status to `new`|
|-|-|

I have manually converted (with a little help from html2rst) the Release Process wiki page into reStructuredText format.

None of the content should have changed, though it is probably in the process I might have munged some text while updating the formatting.

The initial text regarding changes to the doc I placed in a 'warning' declaration, and any obvious notes I replaced with note declarations.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun set status to `closed`|
|-|-|

Considering the release process is still changing, mostly in response to surprises encountered by release managers, and that there are currently known problems with the process, I think that it would be premature to make this change.

In fact, I wonder why we want to do it at all.  The ticket description says "It ought to live near all of the other release essentials." but this doesn't sound like much of a reason.  It's not hard to find the ReleaseProcess wiki page.  It *is* kind of hard to find most of the rest of our documentation though...

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_10"></a>|@hawkowl set status to `reopened`|
|-|-|

I'm reopening this. As our release process implementations change in our source tree (especially with the git migration), it would be nice to have the documented process change with it. For example, [#7985](https://github.com/twisted/twisted/issues/7985) -- it's hard to say what the *new* process would be with any release process changes in a coherent way. If bugs are encountered in the release process, the release ticket can also include the tested alterations to the process.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_11"></a>|@hawkowl commented|
|-|-|

(In [46919]) Branching to release-process-4543.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_12"></a>|@hawkowl commented|
|-|-|

Unfortunately I changed a lot since bramwelt's patch, so I re-rst'd it from scratch.

Builders spun. Please review.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_13"></a>|@adiroiban set owner to @hawkowl<br>@adiroiban set status to `new`|
|-|-|

Thanks. Looks good.

I am happy to have this outside of the wiki and change it together with the release process.

---------

Minor comments.

1.  For the 'What bugs should be considered release blockers?' I would add in that list: ```Any bug which makes the current release worse than the previous one. ```

2. Also, there is a formatting error for that list as all answers are not presented as a list.

3. Same list formatting error for ```Prepare the branch: 4. Make a new quote file for the next version```

Looks good.

Please check my comments and merge.

One this is merge and we have a new release, please don't forget to remove the current wiki page and add a link to the new documentation.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_14"></a>|@hawkowl set status to `closed`|
|-|-|

(In [46931]) Merge release-process-4543: Move the release process docs into the source tree

Author: hawkowl
Reviewer: adiroiban
Fixes: [#4543](https://github.com/twisted/twisted/issues/4543)
